textures/src/main/java/com/dumbdogdiner/stickyapi/util/textures/TextureHelper.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.util.textures;
+
+
+import com.dumbdogdiner.stickyapi.common.util.StringUtil;
+import com.dumbdogdiner.stickyapi.util.http.HttpUtil;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import lombok.experimental.UtilityClass;
+import okhttp3.OkHttpClient;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import org.bukkit.Bukkit;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+/**
+ * Contains various helper methods related to Textures, Texture Strings, and texture validation, as well as providing
+ * access to the bundled texture resources
+ */
+//TODO write javadoc
+@UtilityClass
+public class TextureHelper {
+    // Package-local visibility
+    static final OkHttpClient httpClient = HttpUtil.getDefaultClientInstance();
+
+    private static final Map<String, Map<String, String>> TextureMap = generateTextureMap();
```

Comment:
C-style variable name? Oh no!! I'm guessing this was bc it's static - makes sense?
```suggestion
    private static final Map<String, Map<String, String>> textureMap = generateTextureMap();
```

Reply:
I've seen C style for globals, all caps for constants. Didn't see in styleguide

Reply:
Please don't use C-style for globals if it's not in the style guide.